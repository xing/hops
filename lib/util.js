'use strict';

var fs = require('fs');

var appRoot = require('app-root-path');
var npmlog = require('npmlog');

exports.mergeConfig = function mergeConfig (base, ext) {
  return Object.assign(base || {}, ext, {
    webpack: Object.assign(base ? base.webpack : {}, ext && ext.webpack)
  });
};

exports.getConfig = function readConfig (overrides) {
  var config = {
    webpack: {
      build: require.resolve('../etc/build'),
      develop: require.resolve('../etc/develop'),
      render: require.resolve('../etc/render')
    },
    locations: []
  };
  if (fs.existsSync(appRoot.resolve('package.json'))) {
    exports.mergeConfig(config, appRoot.require('package.json').hops);
  }
  if (fs.existsSync(appRoot.resolve('hops.config.js'))) {
    exports.mergeConfig(config, appRoot.require('hops.config.js'));
  }
  exports.mergeConfig(config, overrides);
  Object.keys(config.webpack).forEach(function (key) {
    if (!fs.existsSync(config.webpack[key])) {
      config.webpack[key] = appRoot.resolve(config.webpack[key]);
    }
  });
  return config;
};

exports.log = function log () {
  npmlog.log.apply(npmlog, ['info', 'hops'].concat(Array.from(arguments)));
};

exports.logError = function logError (err) {
  npmlog.log('error', 'hops', err.stack.toString());
};
