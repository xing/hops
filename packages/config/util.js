var fs = require('fs');
var hopsRoot = require('hops-root');
var hopsConfig = require('.');

var manifestData = null;
var manifestScript = '';

function getRawManifestData () {
  if (!manifestData) {
    manifestData = hopsRoot.require(
      hopsConfig.buildDir,
      'manifest.json'
    );
  }
  return Object.assign({}, manifestData);
}

exports.getManifestScript = function () {
  if (!manifestScript) {
    var filename = getRawManifestData()['manifest.js'];
    if (filename) {
      var filepath = hopsRoot.resolve(hopsConfig.buildDir, filename);
      if (fs.existsSync(filepath)) {
        manifestScript = fs.readFileSync(filepath, 'utf8');
      }
    }
  }
  return manifestScript;
};

exports.getManifestData = function () {
  var manifest = getRawManifestData();
  var keys = Object.keys(manifest).sort(function (a, b) {
    if (a === 'vendor.js') { return -1; }
    if (b === 'vendor.js') { return 1; }
    return a < b ? -1 : a > b ? 1 : 0;
  });
  var assets = { js: [], css: [] };
  return keys.reduce(function (result, key) {
    if (!/^(chunk|manifest).*\.js/.test(key)) {
      if (/\.js$/.test(key)) {
        result.js.push(manifest[key]);
      }
      if (/\.css$/.test(key)) {
        result.css.push(manifest[key]);
      }
    }
    return result;
  }, assets);
};
