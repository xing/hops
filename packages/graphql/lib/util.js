'use strict';

var path = require('path');

var hopsConfig = require('hops-config');

exports.getFragmentsFile = function getFragmentsFile () {
  return path.join(hopsConfig.appDir, 'fragmentTypes.json');
};

exports.getIntrospectionResult = function getIntrospectionResult () {
  var file = exports.getFragmentsFile();
  try {
    require.resolve(file);
    return require(file);
  } catch (_) {}
};
