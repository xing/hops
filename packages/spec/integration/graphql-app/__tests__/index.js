/**
 * @jest-environment jsdom
 */

import { promisify } from 'util';
import React from 'react';
import testRenderer from 'react-test-renderer';
import { HopsTest } from 'hops-test';

import { App } from '../';
import QUERY from '../query.gql';

const tick = promisify(setImmediate);
const flush = async () => {
  await tick();
  await tick();
};

describe('Hops GraphQL', () => {
  it('should render the App correctly', async () => {
    const mockedProviderProps = {
      mocks: [
        {
          request: { query: QUERY },
          result: {
            data: { hello: 'hello test' },
          },
        },
      ],
    };
    const app = testRenderer.create(
      <HopsTest config={{ mockedProviderProps }}>
        <App />
      </HopsTest>
    );
    await flush();

    expect(app).toMatchSnapshot();
  });
});
