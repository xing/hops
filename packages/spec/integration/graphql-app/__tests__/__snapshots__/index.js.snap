// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hops GraphQL should render the App correctly 1`] = `
<h1>
  hello test
</h1>
`;
